### PR TITLE
Issue/46 brightness contrast

### PR DESCRIPTION
**Pull request by joakimlindblad**
Addressing Issue 46 and also Reviewer request for Brightness and Contrast adjustments.

### CODE REVIEW

File: public/js/tmappUI.js [hunk 1]
```diff
@@ -186,6 +191,13 @@ const tmappUI = (function(){
                     annotationTool.complete();
                     break;
                 // ASDF...
+/*                 case 65: // a
+                    tmapp.changeBrightness(+0.1);
+                    break;
+                case 83: // s
+                    tmapp.changeBrightness(-0.1);
+                    break;
+ */
```
Comment (crydell) [inline public/js/tmappUI.js:200]: Can remove this if not used

File: public/js/tmapp.js [hunk 2]
```diff
@@ -80,6 +82,51 @@ const tmapp = (function() {
...
+        _updateBrightnessContrast();
+    }
+
+    /**
+     * Apply _currState brightness/contrast to current viewer
+     */
+    function _updateBrightnessContrast() {
+
+        function contrast_brightness(c,b) {
+            return function(x) {return Math.pow(10,c)*(x-128)+128 + b*128;}
```
Comment (crydell) [inline public/js/tmapp.js:101]: Small detail, but maybe this should be changed in the future to accommodate for images with different color depths.

File: public/index.html [hunk 3]
```diff
@@ -70,6 +71,27 @@
...
+                        <div class="m-2">
+                            <div class="collapse card" id="visualization_collapse">
+                                <div class="card-body">
+                                    <h6 class="card-subtitle text-muted">Brightness</h6>
+                                    <span class="m-2">
+                                        -1 <input id="brightness_slider" type="text" data-slider-min="-1" data-slider-max="1" data-slider-step="0.1" data-slider-value="0"/> &nbsp;1
+                                    </span>
+                                    <h6 class="card-subtitle mt-2 text-muted">Contrast</h6>
+                                    <span class="m-2">
+                                        -1 <input id="contrast_slider" type="text" data-slider-min="-1" data-slider-max="1" data-slider-step="0.1" data-slider-value="0"/> &nbsp;1
```
Comment (crydell) [inline public/index.html:89]: Might want to add a button for quickly resetting all sliders to 0 in the future.

**Review by crydell — APPROVED** [review]
Seems to work as it should. I'm personally getting reduced performance when I'm adjusting the sliders, but I don't think this should be too much of an issue, since slider adjustment might not be going on constantly. I think it's ready to merge.

**Comment by joakimlindblad** [comment]
Agreeing to all comments

**Pull request merged**